### DISCUSSION TITLE
GPU in Sionna

### DISCUSSION THREAD

**Discussion by cacent96** [NVlabs/sionna#92] [Q&A]
Hello,

How is the Sionna module actually using the GPU as a processing device? Is the Sionna module utilizing the GPU capability in processing? What is the component that transfer the data to the GPU for processing and return it to the host if there is no cuda programming involvement?

**Comment by Inv4lidn4m3** [comment]
Sionna is based on Tensorflow (TF). TF is the package that will use the GPU when the neural network is run. The GPU should be automatically used during the training and evaluation of the model if the neural network can be run in graph mode. I do not master all the details, but this is the general idea.

Usage of GPU by TF: https://www.tensorflow.org/guide/gpu
You can refer to the "Performance" section on TF website to have more information on graph and eager mode: https://www.tensorflow.org/guide/function

If you have a common usage of TF (and Sionna), it should work smoothly.
